src/Spago/Build.hs

```diff
@@ -179,42 +179,49 @@ repl cacheFlag newPackages sourcePaths pursArgs depsOnly = do
 -- | Test the project: compile and run "Test.Main"
 --   (or the provided module name) with node
 test :: Spago m => Maybe Purs.ModuleName -> BuildOptions -> [Purs.ExtraArg] -> m ()
-test = runWithNode (Purs.ModuleName "Test.Main") (Just "Tests succeeded.") "Tests failed: "
+test = runBackend Nothing (Purs.ModuleName "Test.Main") (Just "Tests succeeded.") "Tests failed: "
 
 -- | Run the project: compile and run "Main"
 --   (or the provided module name) with node
 run :: Spago m => Maybe Purs.ModuleName -> BuildOptions -> [Purs.ExtraArg] -> m ()
-run = runWithNode (Purs.ModuleName "Main") Nothing "Running failed, exit code: "
+run maybeModuleName buildOpts extraArgs = do
+  Config.Config { alternateBackend } <- Config.ensureConfig
+  runBackend alternateBackend (Purs.ModuleName "Main") Nothing "Running failed, exit code: " maybeModuleName buildOpts extraArgs
 
 -- | Run the project with node: compile and run with the provided ModuleName
 --   (or the default one if that's missing)
-runWithNode
+runBackend
   :: Spago m
-  => Purs.ModuleName
+  => Maybe Text
+  -> Purs.ModuleName
   -> Maybe Text
   -> Text
   -> Maybe Purs.ModuleName
   -> BuildOptions
   -> [Purs.ExtraArg]
   -> m ()
-runWithNode defaultModuleName maybeSuccessMessage failureMessage maybeModuleName buildOpts nodeArgs = do
-  echoDebug "Running NodeJS"
-  outputPath <- getOutputPath buildOpts
-  build buildOpts (Just (nodeAction outputPath))
+runBackend maybeBackend defaultModuleName maybeSuccessMessage failureMessage maybeModuleName buildOpts extraArgs = do
+  echoDebug $ "Running with backend: " <> fromMaybe "nodejs" maybeBackend
+  let postBuild = maybe (nodeAction =<< getOutputPath buildOpts) backendAction maybeBackend
+  build buildOpts (Just postBuild)
   where
     moduleName = fromMaybe defaultModuleName maybeModuleName
-    args = Text.intercalate " " $ map Purs.unExtraArg nodeArgs
-    contents outputPath' =
-         let path = fromMaybe "output" outputPath'
+    nodeArgs = Text.intercalate " " $ map Purs.unExtraArg extraArgs
+    nodeContents = \outputPath'
```

Comment:
It seems to me you didn't start off the most recent master. Yesterday I merged a R fixing bunch of hlint warnings. Could you please modify your PR so that these recent changes are respected? See https://github.com/spacchetti/spago/commit/9737c63b38ca6fbf65455edf2c344c9eec3ab8b9#diff-0b924580c83919e4251597a2dcf79b51L207-L217

E.g. on your PR branch you could do something like:
`git pull --rebase origin master` and then fix the merge conflicts..

Reply:
Fixed. Two hints remaining. Do you want me to fix them too?

<img width="398" alt="Screen Shot 2019-10-10 at 2 37 58 AM" src="https://user-images.githubusercontent.com/67534/66514301-0bfded80-eb07-11e9-8840-02e822809824.png">

Reply:
Thank you! I leave it up to you to decide. Just pick whichever you find more readable :-)